src/dial.js

```diff
@@ -208,7 +213,13 @@ function dial (swtch) {
     }
 
     function openConnInMuxedConn (muxer, cb) {
-      cb(muxer.newStream())
+      muxer.newStream((err, conn) => {
```

Comment:
This asking to change the Stream Muxer interface. https://github.com/libp2p/interface-stream-muxer needs to be updated.

Reply:
I didn't really change `newStream` interface, it already takes a callback as it is. I reverted this to return a connection as before. This PR just tries to propagate the error if it's returned by the multiplexer - before it would just be ignored.